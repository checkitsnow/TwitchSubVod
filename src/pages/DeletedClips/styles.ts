import styled from 'styled-components';

export const Container = styled.div`
  min-height: 100vh;
  display: flex;
  flex: 1;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  padding: 0 2rem;

  form {
    display: flex;
    justify-content: center;
    flex-direction: column;

    .time-container {
      display: flex;
      flex-direction: row;
      justify-content: center;
      align-items: center;

      input {
        width: 2rem;
        margin: 0 0.4rem;
        appearance: textfield;
      }
    }
  }

  input {
    background: var(--light-background);
    height: 2rem;
    width: 20rem;
    padding: 0;
    color: var(--text);
    text-align: center;
    margin-bottom: 0.5rem;
  }

  button[type='submit'] {
    display: flex;
    align-items: center;
    justify-content: center;
    background: var(--purple);
    height: 2rem;
    width: 20rem;
    border: 2px solid var(--purple);
    color: var(--text);

    & svg {
      margin-right: 0.5rem;
    }
  }

  .video-container {
    max-width: 60rem;
    margin-top: 2rem;
  }
`;
