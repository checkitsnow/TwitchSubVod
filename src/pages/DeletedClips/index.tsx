import React, { useState } from 'react';
import axios from 'axios';
import ReactPlayer from 'react-player';

import { FiSearch } from 'react-icons/fi';

import { Container } from './styles';
import LinkBox from '../../components/LinkBox';

const DeletedClips: React.FC = () => {
  const [vodId, setVodId] = useState('');
  const [data, setData] = useState(['']);
  const [startingPointH, setStartingPointH] = useState('');
  const [startingPointM, setStartingPointM] = useState('');
  const [startingPointS, setStartingPointS] = useState('');
  const [endingPointH, setEndingPointH] = useState('');
  const [endingPointM, setEndingPointM] = useState('');
  const [endingPointS, setEndingPointS] = useState('');
  /* const [loading, setLoading] = useState(Number); */

  const hmsToSeconds = (h: any, m: any, s: any) => {
    return +h * 3600 + +m * 60 + +s;
  };

  const handleSubmit = async () => {
    try {
      const startingSeconds = hmsToSeconds(
        startingPointH,
        startingPointM,
        startingPointS,
      );
      const endingSeconds = hmsToSeconds(
        endingPointH,
        endingPointM,
        endingPointS,
      );

      console.log(endingSeconds - startingSeconds);

      if (vodId.length === 11 && endingSeconds - startingSeconds <= 300) {
        for (let i = startingSeconds; i < endingSeconds; i++) {
          // there must be a lighter way to do this (instead of looping through every second)
          let url = `https://twitch-cors.herokuapp.com/https://clips-media-assets2.twitch.tv/${vodId}-offset-${i}.mp4`;
          axios
            .head(`${url}`)
            .then(() => {
              setData((data) => [
                ...data,
                `https://clips-media-assets2.twitch.tv/${vodId}-offset-${i}.mp4`,
              ]);
            })
            .catch(() => {
              return;
            });
        }
      } else {
        throw new Error(
          'Search time is longer than 5 minutes or Vod Id is Invalid',
        );
      }
    } catch (err) {
      console.log(err);
    }
  };

  return (
    <Container>
      <h1>Deleted Twitch Clips</h1>

      <form
        onSubmit={(event) => {
          event.preventDefault();
        }}
      >
        <input
          type="text"
          name="vodId"
          onChange={(event) => setVodId(event.target.value)}
          value={vodId}
          placeholder="Vod ID"
        />
        <div className="time-container">
          <div className="starting-time">
            <input
              type="number"
              name="Starting Point H"
              onChange={(event) => setStartingPointH(event.target.value)}
              value={startingPointH}
              placeholder="h"
              min="0"
            />
            <input
              type="number"
              name="Starting Point M"
              onChange={(event) => setStartingPointM(event.target.value)}
              value={startingPointM}
              placeholder="m"
              min="0"
              max="59"
            />

            <input
              type="number"
              name="Starting Point S"
              onChange={(event) => setStartingPointS(event.target.value)}
              value={startingPointS}
              placeholder="s"
              min="0"
              max="59"
            />
          </div>
          <p>to</p>
          <div className="ending-time">
            <input
              type="number"
              name="Ending Point H"
              onChange={(event) => setEndingPointH(event.target.value)}
              value={endingPointH}
              placeholder="h"
              min="0"
            />
            <input
              type="number"
              name="Ending Point M"
              onChange={(event) => setEndingPointM(event.target.value)}
              value={endingPointM}
              placeholder="m"
              min="0"
              max="59"
            />
            <input
              type="number"
              name="Ending Point S"
              onChange={(event) => setEndingPointS(event.target.value)}
              value={endingPointS}
              placeholder="s"
              min="0"
              max="59"
            />
          </div>
        </div>
        {}
        <button type="submit" onClick={handleSubmit}>
          <FiSearch size={14} />
          Search
        </button>
      </form>

      {/* {loading >= 1 && loading <= 99 &&   <p>{loading}%</p>} */}

      <LinkBox home />

      {data && (
        <>
          {console.log(data)}
          {data.map((item) => (
            <div className="video-container" key={item}>
              <ReactPlayer
                key={item}
                url={item}
                controls
                width="100%"
                height="100%"
              />
            </div>
          ))}
        </>
      )}
    </Container>
  );
};

export default DeletedClips;
