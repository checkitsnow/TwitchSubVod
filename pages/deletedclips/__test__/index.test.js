import React from 'react';
import renderer from 'react-test-renderer';
import DeletedClips from '..';

it('should render Deleted Clips correctly', () => {
  const tree = renderer.create(<DeletedClips />).toJSON();
  expect(tree).toMatchSnapshot();
});
