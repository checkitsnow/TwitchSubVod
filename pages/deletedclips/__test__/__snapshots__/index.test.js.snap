// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Deleted Clips correctly 1`] = `
<div
  className="DeletedClips__Container-sc-18jbdtp-0 bhotBv"
>
  <div
    className="DeletedClips__AnimationContainer-sc-18jbdtp-1 hiaomV"
  >
    <h1>
      Deleted Twitch Clips
    </h1>
    <form
      onSubmit={[Function]}
    >
      <div
        className="input-box"
      >
        <input
          name="vodId"
          onChange={[Function]}
          placeholder="Vod ID"
          type="number"
          value=""
        />
        <div
          className="styles__Container-pgyzn6-0 iSQhKg"
        >
          <div
            className="info-box"
          >
            <p>
              You can get the Vod Id from
              <a
                href="https://twitchtracker.com"
                rel="noopener noreferrer"
                target="_blank"
              >
                twitchtracker.com
              </a>
            </p>
          </div>
          <svg
            fill="none"
            height={18}
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width={18}
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="12"
              cy="12"
              r="10"
            />
            <line
              x1="12"
              x2="12"
              y1="16"
              y2="12"
            />
            <line
              x1="12"
              x2="12.01"
              y1="8"
              y2="8"
            />
          </svg>
        </div>
      </div>
      <div
        className="time-container"
      >
        <div
          className="starting-time"
        >
          <input
            min="0"
            name="Starting Point H"
            onChange={[Function]}
            placeholder="h"
            type="number"
            value=""
          />
          <input
            aria-label="startingM"
            max="59"
            min="0"
            name="Starting Point M"
            onChange={[Function]}
            placeholder="m"
            type="number"
            value=""
          />
          <input
            max="59"
            min="0"
            name="Starting Point S"
            onChange={[Function]}
            placeholder="s"
            type="number"
            value=""
          />
        </div>
        <p>
          to
        </p>
        <div
          className="ending-time"
        >
          <input
            min="0"
            name="Ending Point H"
            onChange={[Function]}
            placeholder="h"
            type="number"
            value=""
          />
          <input
            aria-label="endingM"
            max="59"
            min="0"
            name="Ending Point M"
            onChange={[Function]}
            placeholder="m"
            type="number"
            value=""
          />
          <input
            max="59"
            min="0"
            name="Ending Point S"
            onChange={[Function]}
            placeholder="s"
            type="number"
            value=""
          />
        </div>
      </div>
      <button
        aria-label="submit"
        onClick={[Function]}
        type="submit"
      >
        <svg
          fill="none"
          height={14}
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width={14}
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="11"
            cy="11"
            r="8"
          />
          <line
            x1="21"
            x2="16.65"
            y1="21"
            y2="16.65"
          />
        </svg>
        Search
      </button>
    </form>
    <div
      className="styles__Container-cfnpnh-0 iVQCyu"
    >
      <span
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        <svg
          fill="none"
          height={14}
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width={14}
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="11"
            cy="11"
            r="8"
          />
          <line
            x1="21"
            x2="16.65"
            y1="21"
            y2="16.65"
          />
        </svg>
        Home
      </span>
    </div>
    <div
      className="video-container"
    >
      <div
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
      />
    </div>
  </div>
  <div
    className="styles__Container-w3v8va-0 jGGVKI"
  >
    <a
      href="https://ko-fi.com/pogulive"
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt="Buy Me a Coffee at ko-fi.com"
        className="kofi-img"
        src="https://cdn.ko-fi.com/cdn/kofi5.png?v=2"
      />
    </a>
    <a
      href="https://twitter.com/pogulive"
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt="twitter logo"
        src="https://abs.twimg.com/favicons/twitter.ico"
      />
    </a>
  </div>
</div>
`;
